### PR TITLE
Styling

### PR DESCRIPTION
**Pull request by iSchomer**
I fixed the checked boxes. Within the 'variable name' checkbox, I have made my variable names consistent. With the output list variables, I am debating whether or not I should simplify those lists to an array of lists for each output file. This way I would not have to name and explain output values that I don't know the meaning of; instead I would just describe the array as containing the output information, with its entries in the same order as the output files. Let me know what you think about this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,9 +7,69 @@ See [Cap from the ORNL-CEES team](https://github.com/ORNL-CEES/Cap "Github - ORN
...
+ Here is an example of some basic operations:
+ ```python
+ from pycap import PropertyTree, Charge
+ from battery import *
+ fom energy_storage_device import Dualfoil
+
+ # can run dualfoil sim manually 
+ device = Dualfoil(path='docker/')
+ print(device.get_voltage())
+ # charge for 1 minute
```
Comment (Rombur) [inline README.md]: charge for 1 minutes with a constant current of 10 amperes

File: README.md [hunk 2]
```diff
@@ -7,9 +7,69 @@ See [Cap from the ORNL-CEES team](https://github.com/ORNL-CEES/Cap "Github - ORN
...
+ # see help(pycap) for more details
+ ptree = PropertyTree()
+ ptree.put_double('time_step', 30.0)
+ ptree.put_string('charge_mode', 'constant_current')
+ ptree.put_double('charge_current', 10.0)
+ ptree.put_string('charge_stop_at_1', 'voltage_greater_than')
+ ptree.put_double('charge_voltage_limit', 4.6)
+ ptree.put_bool('charge_voltage_finish', True)
+ ptree.put_double('charge_voltage_finish_max_time',
+                  120)  # 2 minutes after end time
```
Comment (Rombur) [inline README.md]: 2 minutes after end time? what does that mean?

File: README.md [hunk 3]
```diff
@@ -7,9 +7,69 @@ See [Cap from the ORNL-CEES team](https://github.com/ORNL-CEES/Cap "Github - ORN
...
+ ptree.put_double('charge_voltage_finish_current_limit',
+                  5.0)  # or no lower than 5 amps 
+ 
+ from pycap import initialize_data, plot_data
+ data = initialize_data()
+ device.reset()
+ # run the simulation and plot the data
+ cccv = pycap.Charge(ptree)
```
Comment (Rombur) [inline README.md]: cccv? what does that mean?

File: battery/df_manip.py [hunk 4]
```diff
@@ -1,100 +1,146 @@
...
+from datetime import datetime
 
 """
 Module for manipulating the input and output of dualfoil5.1
+
+Note: uses a global variable `file_path` as the path to dualfoil files
 """
 
+# global variable
+file_path = None
```
Comment (Rombur) [inline battery/df_manip.py]: Why do you need to a global variable? global variable are usually a bad idea.

File: README.md [hunk 5]
```diff
@@ -7,9 +7,69 @@ See [Cap from the ORNL-CEES team](https://github.com/ORNL-CEES/Cap "Github - ORN
  +  graphical representation of sample output data generated from Dualfoil5.2
  +  tests to verify the functionality of the `battery` package
 
+## Installation
+ 1. Make sure you have downloaded the latest non-beta version of [docker](https://www.docker.com/products/overview "Get Docker")
```
Comment (dalg24) [inline README.md]: Don't even bother with the instruction on how to download/install of Docker. A working install Docker is a prerequisite.

File: README.md (hunk 1 above)
Comment (dalg24) [inline README.md]: 10 amperes? that sounds like a lot. is that reasonable?

File: battery/df_manip.py [hunk 6]
```diff
@@ -1,100 +1,146 @@
...
+
+                    # skip over all other command lines
+                    while line != '\n':
+                        line = file.readline()
+                    # if next leg depends on time, we need the total time
+                    if (mode != 2) and (mode != -1):
+                        if self.from_restart:
+                            # depends on time AND we are from restart. 
+                            # need total in minutes for dualfoil
+                            totT = get_total_time(self.file_path) / 60
```
Comment (Rombur) [inline battery/df_manip.py]: don't use `totT` I think that you should use `total_time`

File: battery/df_manip.py [hunk 7]
```diff
@@ -103,26 +149,30 @@ def get_total_time(path=''):
     path : str, optional
         Full or relative path to dualfoil files
     """
-    rstFile = open('%sdf_restart.dat' % path, 'r')
+
+    if path is None:
+        rstFile = open('df_restart.dat', 'r')
```
Comment (Rombur) [inline battery/df_manip.py]: Change the name of `rstFile` to follow the style you have chosen

File: battery/df_manip.py [hunk 8]
```diff
@@ -475,29 +517,29 @@ def update_output(self):
...
-        self.heatgen += x[7]
+        x = extract_main_output(path=self.file_path)
+        self.time.extend(x[0])
+        self.n_util.extend(x[1])
+        self.p_util.extend(x[2])
+        self.potential.extend(x[3])
+        self.uocp.extend(x[4])
+        self.curr.extend(x[5])
+        self.temp.extend(x[6])
+        self.heatgen.extend(x[7])
```
Comment (Rombur) [inline battery/df_manip.py]: Use `heat_gen`

File: battery/df_manip.py [hunk 9]
```diff
@@ -475,29 +517,29 @@ def update_output(self):
...
-        self.jside3_prof += x[10]
+        x = extract_profiles(path=self.file_path)
+        self.time_prof.extend(x[0])
+        self.distance_prof.extend(x[1])
+        self.elec_conc_prof.extend(x[2])
+        self.sol_surf_conc_prof.extend(x[3])
+        self.liq_pot_prof.extend(x[4])
+        self.sol_pot_prof.extend(x[5])
+        self.liq_cur_prof.extend(x[6])
+        self.jmain_prof.extend(x[7])
```
Comment (Rombur) [inline battery/df_manip.py]: `j_main_prof` maybe

File: battery/energy_storage_device.py [hunk 10]
```diff
@@ -222,13 +214,18 @@ def evolve_one_time_step_linear_current(self, time_step, current,
...
-                                 path=self.filePath, restart=self.restart)
-            # get next timestep values
-            ttot += ts
-            linCurr += change
-            self.run()
-            self.outbot.update_output()
+            try:
+                self.inbot.add_new_leg(linCurr, ts, 1, "linear current")
+                # get next timestep values and run
+                ttot += ts
```
Comment (Rombur) [inline battery/energy_storage_device.py]: total_time += time_step

File: battery/energy_storage_device.py [hunk 11]
```diff
@@ -222,13 +214,18 @@ def evolve_one_time_step_linear_current(self, time_step, current,
...
+                linCurr += change
```
Comment (Rombur) [inline battery/energy_storage_device.py]: linCurr -> linear_current

File: battery/energy_storage_device.py [hunk 12]
```diff
@@ -252,13 +249,18 @@ def evolve_one_time_step_linear_voltage(self, time_step, voltage,
...
-            # get next timestep values
-            tott += ts
-            linearV += change
-            self.run()
-            self.outbot.update_output()
+            try:
+                self.inbot.add_new_leg(linearV, ts, 0, "linear voltage")
+                # get next timestep values and run
+                tott += ts
+                linearV += change
```
Comment (Rombur) [inline battery/energy_storage_device.py]: same as above

File: battery/energy_storage_device.py [hunk 13]
```diff
@@ -286,13 +288,18 @@ def evolve_one_time_step_linear_power(self, time_step, power,
...
-            # get next timestep values
-            tott += ts
-            linPow += change
-            self.run()
-            self.outbot.update_output()
+            try:
+                self.inbot.add_new_leg(linPow, ts, -2, "linear power")
+                # get next timestep values and run
+                tott += ts
+                linPow += change
```
Comment (Rombur) [inline battery/energy_storage_device.py]: same as above

File: battery/energy_storage_device.py [hunk 14]
```diff
@@ -318,10 +325,15 @@ def evolve_one_time_step_linear_load(self, time_step, load,
...
-            # get next timestep values
-            tott += ts
-            linLoad += change
-            self.run()
-            self.outbot.update_output()
+            try:
+                self.inbot.add_new_leg(linLoad, ts, -3, "linear load")
+                # get next timestep values and run
+                tott += ts
+                linLoad += change
```
Comment (Rombur) [inline battery/energy_storage_device.py]: same as above

File: README.md (hunk 1 above)
Comment (iSchomer) [inline README.md]: The starting voltage for the cell is around 4.53V... what sounds like a more reasonable current to you?

File: README.md [hunk 15]
```diff
@@ -47,29 +46,30 @@ Once you have opened into the Jupyter notebook, you will have access to pycap as
...
  # see help(pycap) for more details
  ptree = PropertyTree()
+ # Set up a simulation that charges until it reaches 4.6V
  ptree.put_double('time_step', 30.0)
  ptree.put_string('charge_mode', 'constant_current')
  ptree.put_double('charge_current', 10.0)
  ptree.put_string('charge_stop_at_1', 'voltage_greater_than')
  ptree.put_double('charge_voltage_limit', 4.6)
+ # sim will hold end voltage until one of the end conditions are met:
+ # 2 minutes of passed OR current falls below 5 amperes
```
Comment (Rombur) [inline README.md]: typo: 2 minutes of passed

**Comment by Rombur** [comment]
Looks good to me. Just fix the typo.

**Pull request merged**